daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Command.scala

```diff
@@ -15,30 +15,79 @@ sealed abstract class Command extends Product with Serializable {
 
 object Command {
 
+  /** Create a template, not by interface */
   final case class Create(
       templateId: Identifier,
       argument: SValue,
   ) extends Command
 
+  /** Create a template, by interface */
+  final case class CreateByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise a template choice, not by interface */
   final case class Exercise(
       templateId: Identifier,
       contractId: SContractId,
       choiceId: ChoiceName,
       argument: SValue,
   ) extends Command
 
+  /** Exercise a template choice, by interface */
+  final case class ExerciseByInterface(
+      interfaceId: Identifier,
+      templateId: Identifier,
+      contractId: SContractId,
+      choiceId: ChoiceName,
+      argument: SValue,
+  ) extends Command
+
+  /** Exercise an interface choice */
+  final case class ExerciseInterface(
+      interfaceId: Identifier,
+      contractId: SContractId,
+      choiceId: ChoiceName,
+      argument: SValue,
+  ) extends Command {
+    // TODO https://github.com/digital-asset/daml/issues/11342
+    //   The actual template id isn't known until run time.
```

Comment:
I don’t understand this. This is not an `APICommand` at least judging from the fact that it doesn’t extend that. So it should only be used for replays and there we should know the  template id.

What am I missing?

Reply:
:+1:

Reply:
You're missing nothing, I didn't realize it wasn't needed. I will remove it.

Reply:
This Speedy command is generated by the commandPreprocessor when invoking `ExerciseCommand` on an interface id directly.

Speedy commands should already know if they are invoking an interface id (generically, like this) or a template id.